### PR TITLE
[skip travis] DOC: 1.6.0 release notes

### PR DESCRIPTION
**Pull request by tylerjereddy**
* add the automatically-generated list of authors for
SciPy `1.6.0`, which will almost certainly require some
`.mailmap` updates

* add the automatically-generated list of issues and PRs
with the `1.6.0` milestone on GitHub

* update the minimum Python version specified in the
release notes to `3.7`, and fix the version number for
the future bug fix release branch

* manually transcribe and reformat the wiki release
notes from:
https://github.com/scipy/scipy/wiki/Release-note-entries-for-SciPy-1.6.0

* using commit message "tags" and draft mode to skip travis CI for now (hopefully)

TODO:

- [x] try to go through as many 1.6.0 PRs labelled with `ENH` as possible & fill in missing ones
- [x] `.mailmap` update to clean up author list a bit?
- [x] finalize "highlights" section?

### CODE REVIEW

File: doc/release/1.6.0-notes.rst [hunk 1]
```diff
@@ -15,66 +15,188 @@ optimizations. Before upgrading, we recommend that users check that
...
 ---------------------------
+`scipy.signal.gammatone` has been added to design FIR or IIR filters that 
+model the human auditory system.
+
+`scipy.signal.iircomb` has been added to design IIR peaking/notching comb 
+filters that can boost/attenuate a frequency from a signal.
 
 
 `scipy.sparse` improvements
 ---------------------------
```
Comment (rgommers) [inline doc/release/1.6.0-notes.rst:135]: Ah this PR just crossed wires with the `csgraph` entry added to the wiki - can you sync that?

Comment (tylerjereddy) [reply doc/release/1.6.0-notes.rst:135]: Yes, note to self--add this:

> A solver for the minimum weight full matching problem for bipartite graphs, also known as the linear assignment problem, has been added in scipy.sparse.csgraph.min_weight_full_bipartite_matching. In particular, this provides functionality analogous to that of scipy.optimize.linear_sum_assignment, but with improved performance for sparse inputs, and the ability to handle inputs whose dense representations would not fit in memory.

**Comment by tylerjereddy** [comment]
Ok, I've pushed a revision to include my changes from a manual scan of the `1.6.0` PRs labelled as enhancements that were not part of the wiki draft of the notes. And various fixes beyond that.

I think there are two things left for the release notes, listed in the `TODO` above. For one of them, @mdhaber @WarrenWeckesser do you want to draft a "highlight" entry I can add for all the `stats` work?

**Comment by ev-br** [comment]
Do you wants us a hand in other TODO items, e.g. mailmap updates?

**Comment by mdhaber** [comment]
There are a bunch of different things, so I suppose a short summary would be:
`scipy.stats` improvements including new distributions, a new test, and enhancements to existing distributions and tests.

But since that's pretty vague, I am fine with just the notes in the stats section, personally.

**Comment by tylerjereddy** [comment]
> Do you wants us a hand in other TODO items, e.g. mailmap updates?

I made a first pass through `.mailmap` updates this evening and updated the author list accordingly as well.

If you do see additional places where an author fix can be made (and you know the author info) do feel free to either push in the change (and maybe use a tag to skip travis ci run) or just comment inline and I'll add it in during a final update of the notes.

**Comment by tylerjereddy** [comment]
I've performed another round of updates for last minute merges this evening. If the CI passes I'm planning to merge this & branch this evening.

**Pull request merged**